File: src/util/main/sp/it/util/file/FileType.java
package sp.it.util.file;

import java.io.File;
import sp.it.util.access.CyclicEnum;

/**
 * Denotes type of [java.io.File]: directory or file.
 */
public enum FileType implements CyclicEnum<FileType> {
	FILE, DIRECTORY;

	public static FileType of(File f) {
		return f.isDirectory() ? DIRECTORY : FILE;
	}
}

File: src/util/main/sp/it/util/file/FileType.kt
package sp.it.util.file

import java.io.File
import org.jetbrains.annotations.Blocking

/** Denotes type of [java.io.File]: directory or file. */
enum class FileType {
   FILE, DIRECTORY;

   companion object {
      @Blocking
      operator fun invoke(f: File): FileType = if (f.isDirectory) DIRECTORY else FILE
   }
}